src/nikonmn_int.cpp

```diff
@@ -3963,4 +3963,20 @@ std::ostream& Nikon3MakerNote::printFocalLd4(std::ostream& os, const Value& valu
   return os;
 }
 
+std::ostream& Nikon3MakerNote::printFocusDistanceLd4(std::ostream& os, const Value& value, const ExifData*) {
+  std::ios::fmtflags f(os.flags());
+  if (value.count() != 1 || value.typeId() != unsignedShort) {
+    os << "(" << value << ")";
+    os.flags(f);
+    return os;
+  }
+  double dist = 0.01 * pow(10.0, (value.toInt64() / 256.0) / 40.0);
+  std::ostringstream oss;
+  oss.copyfmt(os);
+  os << std::fixed << std::setprecision(2) << dist << " m";
+  os.copyfmt(oss);
+  os.flags(f);
```

Comment:
I think `copyfmt` saves the flags too, so I think the `flags` code is redundant.

Reply:
I just did copy/paste of the legacy function, but any new ones indeed omit these, I'll change shortly.